### PR TITLE
doc: add table of supported feature/options tables

### PR DESCRIPTION
**Pull request by xlr-8**
# Issue Type

- Documentation Pull Request

## Summary

To gain a bit in visility about the evolution and/or support of existing
options and/or features 2 tables have been added to make it easier to
see.

Result looks like [this](https://github.com/cristim/autospotting/blob/hr-174/TECHNICAL_DETAILS.md#summary-of-supported-features--options).

Closes: https://github.com/cristim/autospotting/issues/174

## Code contribution checklist

1. [x] The contribution fixes a single existing github issue, and it is linked
   to it.
1. [x] The code is as simple as possible, readable and follows the idiomatic Go
  [guidelines](https://golang.org/doc/effective_go.html).
1. [x] All new functionality is covered by automated test cases so the overall
  test coverage doesn't decrease.
1. [x] No issues are reported when running `make full-test`.
1. [x] Functionality not applicable to all users should be configurable.
1. [x] Configurations should be exposed through Lambda function environment
   variables which are also passed as parameters to the
   [CloudFormation](https://github.com/cristim/autospotting/blob/master/cloudformation/stacks/AutoSpotting/template.json)
   and
   [Terraform](https://github.com/cristim/autospotting/blob/master/terraform/autospotting.tf)
   stacks defined as infrastructure code.
1. [x] Global configurations set from the infrastructure stack level should also
   support per-group overrides using tags.
1. [x] Tags names and expected values should be similar to the other existing
   configurations.
1. [x] Both global and tag-based configuration mechanisms should be tested and
  proven to work using log output from various test runs.
1. [x] The logs should be kept as clean as possible (use log levels as
  appropriate) and formatted consistently to the existing log output.
1. [x] The documentation is updated to cover the new behavior, as well as the
   new configuration options for both stack parameters and tag overrides.
1. [x] A code reviewer reproduced the problem and can confirm the code
  contribution actually resolves it.

### CODE REVIEW

**Comment by xlr-8** [comment]
My understanding is that the support for Beanstalk & CodeDeployed worked, but might need to be better documented as done here https://github.com/cristim/autospotting/pull/200 and reported there https://github.com/cristim/autospotting/issues/18

File: TECHNICAL_DETAILS.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
 
+## Summary of supported features & options ##
+
+| Option  | Status | Override via ASG conf |
+| ------- | :----: | :-------------------: |
+|Run on multiple regions | :white_check_mark: | :heavy_minus_sign: |
+|Keep a fixed minimum percentage of on-demand | :white_check_mark: | :white_check_mark: |
+|Keep a fixed minimum number of on-demand | :white_check_mark: | :white_check_mark: |
+|Can bid at a certain percentage of the on-demand price | :white_check_mark: | :white_check_mark: |
+|Can bid at a higher spot's price | :white_check_mark: | :white_check_mark: |
```
Comment (cristim) [inline TECHNICAL_DETAILS.md]: I'd rephrase this as something like "Can bid the current spot price plus a certain percentage"

File: TECHNICAL_DETAILS.md [hunk 2]
```diff
@@ -1,5 +1,29 @@
 # Technical Details #
 
+## Summary of supported features & options ##
+
+| Option  | Status | Override via ASG conf |
```
Comment (cristim) [inline TECHNICAL_DETAILS.md]: How about some more columns for having the feature exposed via Terraform or CloudFormation, since experimental features may not  be exposed from the start, but just run as command-line options during development.

And also a field for the maturity of each feature

File: TECHNICAL_DETAILS.md [hunk 3]
```diff
@@ -1,5 +1,29 @@
...
+|Can bid at a higher spot's price | :white_check_mark: | :white_check_mark: |
+|Whitelisting of certain instance types | :x: | :heavy_minus_sign: |
+|Blacklisting of certain instance types | :white_check_mark: | :x: |
+
+| Feature | Status |
+| ------- | :----: |
+|Install via Cloudformation | :white_check_mark: |
+|Install via Terraform | :white_check_mark: |
+|Works with code deploy | :white_check_mark: |
+|Works with beanstalk | :white_check_mark: |
```
Comment (cristim) [inline TECHNICAL_DETAILS.md]: I'd like to have some lines added about ECS, Kubernetes, Rancher, Mesos, Nomad, etc.(with links to docs and tools that allow draining the workload)
- Also DefaultVPC, EC2Classic and VPC.

**Comment by xlr-8** [comment]
The branch needs to be updated following the merge of [this PR](https://github.com/cristim/autospotting/pull/204/files). I'll take care of it
